service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/RequestStatementMapper.java

```diff
@@ -58,21 +62,35 @@
 
     private final MessageContext messageContext;
     private final CodeableConceptCdMapper codeableConceptCdMapper;
+    private final ParticipantMapper participantMapper;
 
     public String mapReferralRequestToRequestStatement(ReferralRequest referralRequest, boolean isNested) {
+        final IdMapper idMapper = messageContext.getIdMapper();
         var requestStatementTemplateParameters = RequestStatementTemplateParameters.builder()
-            .requestStatementId(messageContext.getIdMapper().getOrNew(ResourceType.ReferralRequest, referralRequest.getId()))
+            .requestStatementId(idMapper.getOrNew(ResourceType.ReferralRequest, referralRequest.getId()))
             .isNested(isNested)
             .availabilityTime(StatementTimeMappingUtils.prepareAvailabilityTimeForReferralRequest(referralRequest))
             .description(buildDescription(referralRequest))
-            .code(buildCode(referralRequest))
-            .build();
+            .code(buildCode(referralRequest));
 
         if (!referralRequest.hasReasonCode()) {
-            requestStatementTemplateParameters.setDefaultReasonCode(DEFAULT_REASON_CODE_XML);
+            requestStatementTemplateParameters.defaultReasonCode(DEFAULT_REASON_CODE_XML);
         }
 
-        return TemplateUtils.fillTemplate(REQUEST_STATEMENT_TEMPLATE, requestStatementTemplateParameters);
+        if (referralRequest.hasRequester() && referralRequest.getRequester().hasAgent()
+                && referralRequest.getRequester().getAgent().hasReference()) {
+            final String agentRef = idMapper.get(referralRequest.getRequester().getAgent());
```

Comment:
How do we know if the `requester.agent` is a practitioner reference?

Reply:
For my reference:

> if requester.agent is Device reference then generate a string of the form "Requester Device: { human readable Device.code text i.e. term text } " and prepend to RequestStatement/text
> 
> if requester.agent is Organisation reference then generate a string of the form "Requester Org: { Referenced Organisation.name } " and prepend to RequestStatement/text
> 
> if requester.agent is Patient reference then generate a string of the form "Requester: Patient " and prepend to RequestStatement/text
> 
> if requester.agent is RelatedPerson reference then generate a string of the form "Requester: Relation {RelatedPerson.name} " and prepend to RequestStatement/text
> 
> If requester.agent is Practitioner Reference then build a Participant referencing an agentPerson constructed from the requester.agent reference and the requester.onBehalfOf Organization if populated. If there is no requester.onBehalfOfOrganization then fallback to the Organisation on PracttitionerRole or Patient.managingOrganisation
> 
> <Participant typeCode="PRF" contextControlCode="OP"><agentRef classCode="AGNT"><id root="Uuid for participant"/></agentRef></Participant>

Reply:
I have all of the above prepared except for tests and the final point regarding `onBehalfOf`. That one will need some changes made to the agent directory, tbd in another story ∴ blocking this.

Reply:
I also understand it the same way as Natalie, I think we should check here if the agent is the reference to Practitioner and only then generate participant, because only for requester in the type of Practitioner the agentPerson is generated in AgentDirectory.

Reply:
This should now be covered